velox/expression/tests/ExpressionFuzzer.cpp

```diff
@@ -924,26 +961,50 @@ void ExpressionFuzzer::go() {
     // If both paths threw compatible exceptions, we add a try() function to
     // the expression's root and execute it again. This time the expression
     // cannot throw.
-    if (!verifier_.verify(
-            plan,
-            rowVector,
-            resultVector ? BaseVector::copy(*resultVector) : nullptr,
-            true,
-            columnsToWrapInLazy) &&
+    if (verifier_
+            .verify(
+                plan,
+                rowVector,
+                resultVector ? BaseVector::copy(*resultVector) : nullptr,
+                true, // canThrow
+                columnsToWrapInLazy)
+            .exceptionPtr &&
         FLAGS_retry_with_try) {
       LOG(INFO)
           << "Both paths failed with compatible exceptions. Retrying expression using try().";
 
-      plan = std::make_shared<core::CallTypedExpr>(
+      auto tryPlan = std::make_shared<core::CallTypedExpr>(
           plan->type(), std::vector<core::TypedExprPtr>{plan}, "try");
 
       // At this point, the function throws if anything goes wrong.
-      verifier_.verify(
-          plan,
-          rowVector,
-          resultVector ? BaseVector::copy(*resultVector) : nullptr,
-          false,
-          columnsToWrapInLazy);
+      auto tryResult =
+          verifier_
+              .verify(
+                  tryPlan,
+                  rowVector,
+                  resultVector ? BaseVector::copy(*resultVector) : nullptr,
```

Comment:
Instead of reusing resultVector for both runs, if we create a new one , we can compare the  try result and the results without try for non null rows (by wrapping try result in non null indices). This will ensure that across all runs values are same.